brave/brave-core
# feat(wallet): Balance Details Modal

## Description 
Introduces a `Balance Details Modal` that is viewable when a `Bitcoin` account has a pending balance.


Resolves <https://github.com/brave/brave-browser/issues/35258>



## Submitter Checklist:

  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:

1. Create 2 `Bitcoin Testnet` accounts
2. Fund one of your account from a `Bitcoin Testnet Faucet`
3. Send some of your `Testnet Bitcoin` to your other account.
4. Go the `Accounts` tab and view details for that account
5. You should see an `Info Bar` stating that `Some balances may be unavailable`
6. Click on `View details`, you should then see the `Balance Details Modal`
7. Go to the `Portfolio` tab and sort by accounts. Follow the same steps above.
8. Go to the `Send` page and select to send your `Testnet Bitcoin` from the account you are receiving.
9. You should see a warning on the `Send` page displaying the amount that is available.
10. Click on details and you should see the `Balance Details Modal`.

## Wallet Page:

![Screenshot 23](https://github.com/brave/brave-core/assets/40611140/f76190a6-1111-4638-aef5-e95a5765fac2)

![Screenshot 22](https://github.com/brave/brave-core/assets/40611140/691a0dcb-fe31-425f-9874-9624ceb506d1)

![Screenshot 24](https://github.com/brave/brave-core/assets/40611140/a29b8bf4-4adc-45ac-bfa7-92463e6674c2)

![Screenshot 25](https://github.com/brave/brave-core/assets/40611140/ed0959a0-190d-47cb-9a86-bb2c1935a465)

## Wallet Panel:

![Screenshot 21](https://github.com/brave/brave-core/assets/40611140/f5d9b03a-3dbe-4242-871f-3ba3ea408b0c)

![Screenshot 20](https://github.com/brave/brave-core/assets/40611140/2b6477de-6dbf-4b20-9cdd-c4c920174ed4)

![Screenshot 26](https://github.com/brave/brave-core/assets/40611140/bdd6340c-843e-4180-9bf8-ede70281757d)

![Screenshot 27](https://github.com/brave/brave-core/assets/40611140/6c1fb7cb-60ed-4ca1-b12d-483ab3ec5e91)
